std/node/_stream/end-of-stream.ts

```diff
@@ -0,0 +1,247 @@
+import { once } from "../_utils.ts";
+import type Readable from "./readable.ts";
+import type Writable from "./writable.ts";
+import {
+  ERR_INVALID_ARG_TYPE,
+  ERR_STREAM_PREMATURE_CLOSE,
+  NodeErrorAbstraction,
+} from "../_errors.ts";
+
+type Stream = Readable | Writable;
+
+function isRequest(stream: Stream) {
+  // deno-lint-ignore ban-ts-comment
+  //@ts-ignore
+  return stream.setHeader && typeof stream.abort === "function";
+}
+
+function isReadable(stream: Stream) {
```

Comment:
```suggestion
function isReadable(stream: Stream): stream is Readable {
```

Reply:
Yes and no. Node implementation is a mess. Even if it does't extend Readable per se, any object can qualify as such as long as it has properties in the function enabled (Duplex, Sockets and more do this).